optimum/neuron/distributed/checkpointing.py

```diff
@@ -16,32 +16,58 @@
 
 import json
 from pathlib import Path
-from typing import Dict, Literal, Union
+from typing import Any, Callable, Dict, List, Literal, Union
 
 import torch
 from transformers.modeling_utils import shard_checkpoint
 from transformers.utils import SAFE_WEIGHTS_INDEX_NAME, SAFE_WEIGHTS_NAME, WEIGHTS_INDEX_NAME, WEIGHTS_NAME
 
-from ..utils.require_utils import requires_safetensors
-from .utils import TENSOR_PARALLEL_SHARDS_DIR_NAME, ParameterMetadata
+from ..utils.require_utils import requires_neuronx_distributed, requires_safetensors
+from .utils import TENSOR_PARALLEL_SHARDS_DIR_NAME, ParameterMetadata, compute_query_indicies_for_rank
 
 
-def consolidate_tensor_parallel_checkpoints(checkpoint_dir: Union[str, Path]) -> Dict[str, "torch.Tensor"]:
-    if not isinstance(checkpoint_dir, Path):
-        checkpoint_dir = Path(checkpoint_dir)
-
-    if checkpoint_dir.name != TENSOR_PARALLEL_SHARDS_DIR_NAME:
-        if (checkpoint_dir / TENSOR_PARALLEL_SHARDS_DIR_NAME).is_dir():
-            checkpoint_dir = checkpoint_dir / TENSOR_PARALLEL_SHARDS_DIR_NAME
-        else:
-            raise ValueError(f"Could not find the tensor parallel shards from {checkpoint_dir}")
-
+def create_gqa_query_or_output_projection_weight_from_full_weight(
+    full_weight: torch.Tensor,
+    tp_size: int,
+    num_attention_heads: int,
+    num_key_value_heads: int,
+    kv_size_multiplier: int,
+    query_or_output: Union[Literal["query"], Literal["output"]],
+):
+    assert query_or_output in ["query", "output"]
+    assert full_weight.device == torch.device("cpu")
```

Comment:
Maybe raise with information?

Reply:
I do not think it's needed as it's quite low level functions.